cmd/kubeadm/app/cmd/config.go

```diff
@@ -353,7 +353,7 @@ func newCmdConfigImages(out io.Writer) *cobra.Command {
 func newCmdConfigImagesPull() *cobra.Command {
 	externalClusterCfg := &kubeadmapiv1beta2.ClusterConfiguration{}
 	kubeadmscheme.Scheme.Default(externalClusterCfg)
-	externalInitCfg := &kubeadmapiv1beta2.InitConfiguration{}
+	externalInitCfg := cmdutil.DefaultInitConfiguration()
```

Comment:
so `config images pull` checks the socket and either uses `docker` or `crictl`:
https://github.com/kubernetes/kubernetes/blob/95198130d78d253b589b0ba9076d60f155f412bc/cmd/kubeadm/app/cmd/config.go#L373

this change would enforce usage of `crictl`, which might break some setups.
i think we should omit this change and let the socket detection handle it.